woocommerce/woocommerce
# Product Gallery block: Prevent block error when a user inputs a decimal number for the Number of Thumbnails setting

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR prevents an error when a user inputs a decimal number in the Number of Thumbnails settings of the Product Gallery block. Now, every time the input changes, we round it to the nearest integer (for instance, 4.2 would be converted to 4, while 4.8 would be converted to 5).

E2E tests were also included to cover this functionality and prevent future errors.

Closes #43796.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Single Product template (Appearance > Editor > Templates > Single Product template).
2. Click on "Edit" to open the WordPress editor.
3. In the editor, you should see a block inserter icon or an option to "Add Block." Click on it.
4. In the block inserter, search for "Product Gallery" or browse the available blocks until you find them.
5. Click on the "Product Gallery" block to add it to your content.
6. Select the "Product Gallery (Beta)" block.
7. On the right-side menu, change the "Number of Thumbnails" input field to some decimal value, for example 4.2.
8. Make sure you see four thumbnails being rendered.
9. On the right-side menu, change the "Number of Thumbnails" input field to some different decimal value, such as 5.6.
10. Make sure you see six thumbnails being rendered.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Restrict the Number of Thumbnails setting from the Product Gallery block to integer values

#### Comment 

</details>
